astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+        # Add trailing single whitespaces to Bytes column for better visibility.
+        singlebfmt += " "
+        fmtb += " "
+
+        # Set default width of Label and Description Byte-By-Byte columns.
+        max_label_width, max_descrip_size = 7, 16
+
+        bbb = Table(names=['Bytes', 'Format', 'Units', 'Label', 'Explanations'],
+                    dtype=[str]*5)
+
+        # Iterate over the columns to write Byte-By-Byte rows.
+        for i, col in enumerate(self.cols):
+            # Check if column is MaskedColumn
+            col.has_null = isinstance(col, MaskedColumn)
+
+            # Check if the column format was set by the passed ``formats`` argument.
+            if col.format is None:
+                if col.info.format is not None:
+                    col.format = col.info.format
+
+            if col.format is not None:
+                col.meta.size = max([len(sval) for sval in col.str_vals])
+
+            # Set CDSColumn type, size and format.
+            if np.issubdtype(col.dtype, np.dtype(int).type) or np.issubdtype(col.dtype, np.integer):
+                # Integer formatter
+                self._set_column_val_limits(col)
+                if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                    col.meta.size = max(len(str(col.max)), len(str(col.min)))
+                col.fortran_format = "I" + str(col.meta.size)
+                col.format = ">" + col.fortran_format[1:]
+
+            elif np.issubdtype(col.dtype, np.dtype(float).type):
+                # Float formatter
+                self._set_column_val_limits(col)
+                self.column_float_formatter(col)
+
+            else:
+                # String formatter, ``np.issubdtype(col.dtype, str)`` is ``True``.
+                dtype = col.dtype.str
+                if col.has_null:
+                    mcol = col
+                    mcol.fill_value = ""
+                    coltmp = Column(mcol.filled(), dtype=str)
+                    dtype = coltmp.dtype.str
+                if getattr(col.meta, 'size', None) is None:  # If ``formats`` not passed.
+                    col.meta.size = int(re.sub(r'^[^0-9]+(\d+)$', r'\1', dtype))
+                col.fortran_format = "A" + str(col.meta.size)
+                col.format = str(col.meta.size) + "s"
+
+            endb = col.meta.size + startb - 1
+
+            # ``mixin`` columns converted to string valued columns will not have a name
+            # attribute. In those cases, a ``Unknown`` column label is put, indicating that
+            # such columns can be better formatted with some manipulation before calling
+            # the CDS/MRT writer.
+            if col.name is None:
+                col.name = "Unknown"
+
+            # Set column description.
+            if col.description is not None:
+                description = col.description
+            else:
+                description = "Description of " + col.name
```

Comment:
I think the default description if not provided should just be blank, or perhaps `---` or the column name  if there must be something there. The `Description of ...` is just adding noise but not information.

Reply:
This strongly encourages users to put in a description for every column though, which I think is the intend of the MRT format. MRT tables will (in the use case that we are working towards, although I'm use users will find a way to abuse that) always be edited by journal staff to put in the proper reference information of the published article, so an pretty obvious "PUT SOEMTHING HERE" can be useful. That said,  `---` fills the same role and is a little less obtrusive.

Reply:
Since, the output of the tests will change when this is done, it would be better to address this separately. So, **needs an issue**.